concourse/pipelines/create-cloudfoundry.yml

```diff
@@ -1798,6 +1798,100 @@ jobs:
+              type: docker-image
+              source:
+                repository: governmentpaas/cf-uaac
+                tag: 895cf6752c8ec64af05a3a735186b90acd3db65a
+            params:
+              DISABLE_METRICS_USER_CREATION: ${DISABLE_METRICS}
+            inputs:
+              - name: paas-cf
+              - name: cf-secrets
+              - name: bosh-CA
+              - name: cf-manifest
+            run:
+              path: sh
+              args:
+                - -e
+                - -u
+                - -c
+                - |
+                  if [ "${DISABLE_METRICS_USER_CREATION:-}" = "true" ]; then
+                    echo "Metrics user creation is disabled (DISABLE_METRICS_USER_CREATION=${DISABLE_METRICS_USER_CREATION}). Skipping."
+                    exit 0
+                  fi
+                  ./paas-cf/concourse/scripts/import_bosh_ca.sh
+
+                  VAL_FROM_YAML=$(pwd)/paas-cf/concourse/scripts/val_from_yaml.rb
+                  UAA_ADMIN_CLIENT_PASS=$($VAL_FROM_YAML secrets.uaa_admin_client_secret cf-secrets/cf-secrets.yml)
+                  UAA_ENDPOINT=$($VAL_FROM_YAML properties.uaa.url cf-manifest/cf-manifest.yml)
+                  METRICS_PASSWORD=$($VAL_FROM_YAML secrets.paas_metrics_password cf-secrets/cf-secrets.yml)
+                  METRICS_USERNAME=paas-metrics
+
+                  uaac target "${UAA_ENDPOINT}"
+                  uaac token client get admin -s "${UAA_ADMIN_CLIENT_PASS}"
+
+                  if uaac user get "${METRICS_USERNAME}" -a id >/dev/null; then
+                    echo "user ${METRICS_USERNAME} already exists ...skipping"
+                    exit 0
+                  fi
+
+                  echo "creating user ${METRICS_USERNAME}"
+                  uaac user add "${METRICS_USERNAME}" -p "${METRICS_PASSWORD}" --emails ignored
+                  USERID=$(uaac user get "${METRICS_USERNAME}" -a id | awk '{print $2}')
+
+                  set +e
+                  add_member(){
+                    GID=$(uaac group get $1 -a id | awk '{print $2}')
+                    result=$(
+                      uaac curl \
+                        -k "/Groups/${GID}/members" \
+                        -XPOST \
+                        -H 'Content-Type: application/json' \
+                        -d '{"origin":"uaa","type":"USER","value":"'${2}'"}'
+                    )
+                    # Check that `uaac` output a 201 HTTP status code.
+                    # N.B., It stopped printing `201 Created` during the upgrade to
+                    # CF v269, and now just prints the `201` code on its own line
+                    # followed by a space.
+                    if [ $result = 0 ] && echo "${result}" | grep '^201 $'; then
+                      echo "${result}"
+                      echo "Added $2 to $1"
+                    else
+                      echo "${result}"
+                      exit 1
+                    fi
+                  }
```

Comment:
Yes I think you're right a standard "create user" script is probably a good idea.

...but I don't feel confident messing with the create_admin script to factor it out as part of this changeset :)